internal/service/common/device.go

```diff
@@ -2,18 +2,29 @@ package common
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"strings"
 	"time"
 
+	api "github.com/flightctl/flightctl/api/v1alpha1"
 	"github.com/flightctl/flightctl/internal/api/server"
 	"github.com/flightctl/flightctl/internal/flterrors"
 	"github.com/flightctl/flightctl/internal/store"
+	"github.com/flightctl/flightctl/internal/util"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
 )
 
-func ReplaceDeviceStatus(ctx context.Context, st store.Store, request server.ReplaceDeviceStatusRequestObject) (server.ReplaceDeviceStatusResponseObject, error) {
+func ReplaceDeviceStatus(ctx context.Context, st store.Store, log logrus.FieldLogger, request server.ReplaceDeviceStatusRequestObject) (server.ReplaceDeviceStatusResponseObject, error) {
 	orgId := store.NullOrgId
 
 	device := request.Body
+	if errs := validateDeviceStatus(device); len(errs) > 0 {
+		return server.ReplaceDeviceStatus400JSONResponse{Message: errors.Join(errs...).Error()}, nil
```

Comment:
This bit me the other day but we need to be careful on this usage because if someone by mistake returns nil for []error it will panic here. Not sure what the action item is here but we should consider this issue with this pattern. I guess we could nill check err but not as clean.


```
	var errs []error
	errs = nil
	err := errors.Join(errs...).Error()
	fmt.Println(err)
```

Reply:
If `errs == nil`, then `len(errs) == 0`, so we should catch that case, or?

Reply:
haha my bad tunnel vision I missed the upper check.